### PR TITLE
Feature: Add `analytics.getEnrollmentsQuery` method

### PR DESCRIPTION
**Pull request by delemaf**
Following [Zebra program indicator feature](https://app.clickup.com/t/8694tp0tp) we needed to get enrollments query in analytics `/analytics/enrollments/query/`
- New method `analytics.getEnrollmentsQuery`

#8694tp0tp

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-8694tp0tp Create Event List - National ](https://app.clickup.com/t/8694tp0tp)

File: README.md [hunk 1]
```diff
@@ -156,6 +156,17 @@ const analyticsData = await api.analytics
...
 
+#### Get enrollments query
+
+```
+const analyticsData = await api.analytics
+    .getEnrollmentsQuery("Gs69Uw2Mom1", {
+        dimension: ["cYeuwXTCPkU", "fbfJHSPpUQD"],
+        enrollmentDate: "LAST_12_MONTHS,THIS_MONTH",
+    })
+    .getData();
```
Comment (tokland) [inline README.md:167]: In the README, we typically include an example that would work with https://play.dhis2.org/dev/

File: src/api/analytics.ts [hunk 2]
```diff
@@ -54,7 +57,11 @@ export type AnalyticsResponse = {
         hidden: boolean;
         meta: boolean;
     }>;
-
+    metaData: {
+        dimensions: Record<string, string[]>;
+        items: Record<string, { name: string; uid?: Id; code?: string; options: any[] }>;
+        pager: Pager;
+    };
```
Comment (tokland) [inline src/api/analytics.ts:64]: when `skipMeta=true`, this object is empty, so : {} | {...}. Also, I didn't get a pager on test request I did, so let's put it optional.

It's hard to get the type exact, as the options influence the shape of the response, but for now this should be enough.

File: src/api/analytics.ts [hunk 3]
```diff
@@ -87,6 +94,16 @@ export class Analytics {
         return this.d2Api.get<AnalyticsResponse>("/analytics", options);
     }
 
+    getEnrollmentsQuery(
+        programId: Id,
+        options: AnalyticsOptions
+    ): D2ApiResponse<AnalyticsResponse> {
```
Comment (tokland) [inline src/api/analytics.ts:100]: When definining the signature of a function, the most important is how it looks like being called: `getEnrollmentsQuery("Gs69Uw2Mom1")`. But what is the id? I'd create a GetEnrollmentsQueryOptions with AnalyticsOptions + programId. Then the call looks: `getEnrollmentsQuery({programId: "Gs69Uw2Mom1"})`.

File: README.md (hunk 1 above)
Comment (tokland) [reply README.md:167]: Non-related comment: also, in analytics.ts, add links to the section of the docs relevant. Example: https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-240/analytics.html#webapi_enrollment_analytics

**Review by tokland — APPROVED** [review]
[code only review] all good

**Review by ifoche — APPROVED** [review]
Thanks @delemaf

**Pull request merged**